packages/offix-datastore/src/DataStore.ts

```diff
@@ -35,30 +35,36 @@ export interface CustomEngines {
  * and GraphQL replication engine.
  */
 export class DataStore {
-  private models: Model<unknown>[];
   private storage: LocalStorage;
+  private replicator?: IReplicator;
 
   private config: DataStoreConfig;
-  private engines: CustomEngines | undefined;
 
   constructor(config: DataStoreConfig, engines?: CustomEngines) {
     this.config = config;
-    this.models = [];
-    this.engines = engines;
 
     if (engines && engines.storeAdapter) {
       this.storage = new LocalStorage(engines.storeAdapter);
     } else {
       const indexedDB = new IndexedDBStorageAdapter();
       this.storage = new LocalStorage(indexedDB);
     }
+
+    if (this.config.replicationConfig) {
```

Comment:
-1 for that. 

There are async methods there. 
I think we might need to have init with await and we cannot have this code in constructor.